pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -22,12 +27,18 @@ func validateNetworking(networking *mesh_proto.Dataplane_Networking) validators.
 	if len(networking.GetInbound()) > 0 && networking.Gateway != nil {
 		err.AddViolationAt(path, "inbound cannot be defined both with gateway")
 	}
+	// backwards compatibility validate networking.address only when all inbounds are in new format
+	if !hasLegacyInbound(networking) {
```

Comment:
Could we please change this condition to

```
if networking.Address != "" || !hasLegacyInbound(networking) {
    ...
}
```

?

Reply:
done